src/layouting/data_limits.jl

```diff
@@ -230,3 +228,20 @@ function data_limits(plot::Image)
     maxi = Vec3f(last.(mini_maxi)..., 0)
     return Rect3f(mini, maxi .- mini)
 end
+
+function bbox_mesh(coords)
+    dim = length(coords[1])
+    mini = [xyz for xyz in coords[1]]
+    maxi = [xyz for xyz in coords[1]]
+    for coord in coords
+        for d in 1:dim
+            if coord[d] < mini[d]
+               mini[d] = coord[d]
+            elseif coord[d] > maxi[d]
+               maxi[d] = coord[d]
+            end
+        end
```

Comment:
Would be good to account for NaN and Inf values here.  Ideally we would ignore those.

Reply:
If we assume at this point that coords are allowed to be nan or inf, then the initialization of mini and maxi is broken, because the first coordinate could be nan or inf.

Reply:
That seems to be a limitation of the current implementation too, now that you mention it :(

https://github.com/MakieOrg/Makie.jl/blob/84a1cdf49d718f19b6fee36978c265fa1ad3ab03/src/layouting/data_limits.jl#L187-L192
https://github.com/MakieOrg/Makie.jl/blob/84a1cdf49d718f19b6fee36978c265fa1ad3ab03/src/layouting/data_limits.jl#L174-L185

Reply:
Coordinates may not be provided by the user as NaN or Inf, but transformation functions may force them to be. 

here's a potential solution:
```julia
mini = [typemax(xyz) for xyz in coords[1]]
maxi = [typemin(xyz) for xyz in coords[1]]
```
with the caveat that it will return infinite bounding boxes if there are no points which are not NaN.

I guess we could also look at finding the first finite point using `findfirst(isfinite, coords)` and going from there.